mlir/include/mlir/Dialect/Rock/Tuning/Serializable.h

```diff
@@ -71,21 +71,12 @@ struct Serializable {
   }
 
   bool checkVersionFormat(const std::string &s) {
-    int32_t maxNumTokens;
-    switch (version) {
-    case Version::V1:
-      maxNumTokens = 8;
-      break;
-    case Version::V2:
-      maxNumTokens = 9;
-      break;
-    case Version::V3:
-      maxNumTokens = 11;
-      break;
-    default:
+    const int32_t maxNumTokensArray[] = {0, 8, 9, 11, 12};
```

Comment:
We should have some unit-tests for the serializing perfConfigs. Now that we have many versions,  it is a good idea to make sure they are being parsed/deserialized correctly.